cmd/gateway/main.go

```diff
@@ -216,18 +218,53 @@ func (flags GatewayFlags) action(ctx context.Context, cliCtx *cli.Context, ident
 }
 
 // NewGateway creates a new minio Gateway
-func (flags GatewayFlags) NewGateway(ctx context.Context, identity *identity.FullIdentity) (gw minio.Gateway, err error) {
-	metainfo, streams, err := flags.GetMetainfo(ctx, identity)
+func (flags GatewayFlags) NewGateway(ctx context.Context, ident *identity.FullIdentity) (gw minio.Gateway, err error) {
+	key := new(storj.Key)
+	copy(key[:], flags.Enc.Key)
+
+	uplink, err := libuplink.NewUplink(ctx, &libuplink.Config{
+		Volatile: struct {
```

Comment:
This pattern is really fragile- breaks any time someone adds or changes a field in the Volatile struct in question, which will presumably happen sometimes because it's "Volatile". Probably easier/less code/less fragile to instantiate an empty libuplink.Config and then set the values you care about?

Reply:
yeah i think i've decided i hate it. we could also make a named type for the Volatile substruct instead of unnamed structs